src/librbd/ExclusiveLock.cc

```diff
@@ -100,6 +101,10 @@ void ExclusiveLock<I>::init(uint64_t features, Context *on_init) {
     ML<I>::set_state_initializing();
   }
 
+  on_init = new FunctionContext([this, on_init](int r) {
```

Comment:
Nit: create a helper ```C_RefCountedRequest``` so you can avoid creating lots of ```FunctionContext``` lambdas that do the identical thing all over the place.